client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx

```diff
@@ -1,26 +1,6 @@
 /** @format */
-
-jest.mock( 'lib/abtest', () => ( {
-	abtest: () => '',
-} ) );
-
-jest.mock( '../google-vouchers', () => ( {} ) );
-jest.mock( '../video-audio-posts', () => {
-	const React = require( 'react' );
-	return class VideoAudioPosts extends React.Component {};
-} );
-
-jest.mock( 'i18n-calypso', () => ( {
-	localize: Comp => props => (
-		<Comp
-			{ ...props }
-			translate={ function( x ) {
-				return x;
-			} }
-		/>
-	),
-	numberFormat: x => x,
-} ) );
+jest.mock( '../google-vouchers', () => 'GoogleVouchers' );
+jest.mock( '../video-audio-posts', () => 'VideoAudioPosts' );
```

Comment:
Using a string to replace a component is… interesting. It's mostly fine in this case, however is a bit limiting in that these aren't valid elements. A nice workaround is to replace the `UpperCamelCase` with `kebab-case`, this will trick React so that it believes it's rendering custom elements, and everything just works. The downside is the tests that `find( 'VideoAudioPosts' )` would need to find the mocked string.

Just FYI, I'm not sold on the best approach with these string mocks.

Reply:
What's the downside here? `find('GoogleVouchers')` will work just fine

Reply:
I'm not sure how `shallow` avoids this, it may not rely on React at all. However, If React is passed one of these string-mocks, for example via enzyme's `render`, it will warn because UpperCamelCase should be React components.

>     console.error node_modules/fbjs/lib/warning.js:33
>      Warning: <GoogleVouchers /> is using uppercase HTML. Always use lowercase HTML tags in React.

```diff
diff --git a/client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx b/client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
index cffc32a0ca..730a56a0c8 100644
--- a/client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
+++ b/client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
@@ -3,7 +3,7 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
+import { render, shallow } from 'enzyme';
 import React from 'react';
 import {
 	PLAN_FREE,
@@ -250,7 +250,7 @@ describe( 'ProductPurchaseFeaturesList feature functions', () => {
 	} );
 
 	test( 'getPremiumFeatures() should pass proper plan type to VideoAudioPosts child component', () => {
-		const comp = shallow( <ProductPurchaseFeaturesList { ...props } plan={ PLAN_PREMIUM } /> );
+		const comp = render( <ProductPurchaseFeaturesList { ...props } plan={ PLAN_PREMIUM } /> );
 		const audioPosts = comp.find( 'VideoAudioPosts' );
 		expect( audioPosts ).toHaveLength( 1 );
 		expect( audioPosts.prop( 'plan' ) ).toBe( PLAN_PREMIUM );
```

As I say, fine in this case, but has the potential for leading to peculiar behavior in the future.